frontend/src/main/scala/bloop/monix/FoldLeftAsyncConsumer.scala

```diff
@@ -0,0 +1,83 @@
+package bloop.monix
+
+import monix.eval.{Callback, Task}
+import monix.execution.Ack.{Continue, Stop}
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.execution.cancelables.{AssignableCancelable, CompositeCancelable}
+import monix.execution.misc.NonFatal
+import monix.reactive.{Consumer, Observable}
+import monix.reactive.observers.Subscriber
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Future
+
+// Fork of `FoldLeftAsyncConsumer` from Monix
+final class FoldLeftAsyncConsumer[A, R](
```

Comment:
What are the differences with the reference implementation from Monix? Why can't we use it?

Reply:
https://github.com/scalacenter/bloop/pull/389/files/3b07914214c27039658685c68355de169f083d3c#diff-935b4e27e7e2af3e3f5402c4e6b9bcc4R62 and the implementation of `apply` are the differences. Essentially, we add cancellation logic to this consumer.